source/MaterialXGenShader/OCIOColorManagementSystem.cpp

```diff
@@ -0,0 +1,634 @@
+                        ValuePtr uniformValue = uniformItem->_value;
+                        InputPtr newInput = impl->addInput(rmapItem.first, uniformValue->getTypeString());
+                        if (newInput)
+                        {
+                            newInput->setValue(uniformValue->getValueString(), uniformValue->getTypeString());
+                            newInput->setIsUniform(true);
+                        }
+                    }
+                }
+                else
+                {
+                    for (auto rmapItem : *rmap)
+                    {
+                        ColorSpaceTexturePtr uniformItem = std::static_pointer_cast<ColorSpaceTexture>(rmapItem.second);
+                        const FloatVec& vecarray = uniformItem->data;
+                        ValuePtr uniformValue = Value::createValue(vecarray);
+                        InputPtr newInput = impl->addInput(rmapItem.first, uniformValue->getTypeString());
+                        if (newInput)
+                        {
+                            newInput->setValue(uniformValue->getValueString(), uniformValue->getTypeString());
+                            newInput->setAttribute("width", std::to_string(uniformItem->width));
+                            newInput->setAttribute("height", std::to_string(uniformItem->height));
+                            newInput->setAttribute("depth", std::to_string(uniformItem->depth));
+                            newInput->setAttribute("channels", std::to_string(uniformItem->channelCount));
+                            newInput->setIsUniform(true);
+                        }
+                    }
+                }
+            }
+        }
+        return impl;
+    }
+    catch (const OCIO::Exception& e)
+    {
+        throw ExceptionShaderGenError("Code generation failed for transform: '" + 
+            transform.sourceSpace + "' to '" + transform.targetSpace + "'. Type: '" + transform.type->getName()  + "' : " + e.what());
+    }
+
+    return nullptr;
+}
+
+const ColorManagementResourceMapPtr OCIOColorManagementSystem::getResource(ResourceType resourceType) const
+{
+    if (_ocioInfo)
+    {
+        return _ocioInfo->resourceMap[(int)resourceType];
+    }
+    return nullptr;
+}
+
+void OCIOColorManagementSystem::clearResources()
+{
+    if (_ocioInfo)
+    {
+        _ocioInfo->resourceMap->reset();
+    }
+}
+
+
+//
+// OCIO source code node method
+//
+
+OCIOSourceCodeNode::OCIOSourceCodeNode()
+    : _cmUniforms(ColorManagementSystem::COLOR_MANAGEMENT_UNIFORMS, EMPTY_STRING)
+{
+}
+
+ShaderNodeImplPtr OCIOSourceCodeNode::create()
+{
+    return std::make_shared<OCIOSourceCodeNode>();
+}
+
+void OCIOSourceCodeNode::initialize(const InterfaceElement& element, GenContext& context)
+{
+    SourceCodeNode::initialize(element, context);
+
+    // Cache a uniform block
+    for (auto nodeInput : element.getInputs())
```

Comment:
Init just caches information to use when the node is called for later calls to emit code, create uniforms etc.